daemon/api_notices.go

```diff
@@ -232,6 +246,104 @@ func allowedNoticeTypesForInterface(iface string) []state.NoticeType {
 	return types
 }
 
+func postNotices(c *Command, r *http.Request, user *auth.UserState) Response {
+	requestUID, err := uidFromRequest(r)
+	if err != nil {
+		return Forbidden("cannot determine UID of request, so cannot create notice")
+	}
+
+	var payload struct {
+		Action string `json:"action"`
+		Type   string `json:"type"`
+		Key    string `json:"key"`
+		// NOTE: Data and RepeatAfter fields are not needed for snap-run-inhibit notices.
+	}
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&payload); err != nil {
+		return BadRequest("cannot decode request body: %v", err)
+	}
+
+	if payload.Action != "add" {
+		return BadRequest("invalid action %q", payload.Action)
+	}
+	if payload.Type != "snap-run-inhibit" {
+		return BadRequest(`invalid type %q (can only add "snap-run-inhibit" notices)`, payload.Type)
+	}
+	if len(payload.Key) > maxNoticeKeyLength {
+		return BadRequest("key must be %d bytes or less", maxNoticeKeyLength)
+	}
+
+	st := c.d.overlord.State()
+	st.Lock()
+	defer st.Unlock()
+
+	fromSnapCmd, err := requestFromSnapCmd(st, r)
+	if err != nil {
+		return InternalError("cannot check request source")
+	}
+	if !fromSnapCmd {
+		return Forbidden("only snap command can record notices")
+	}
```

Comment:
Since they aren't used further down, we can scope the variables here
```suggestion
	if fromSnapCmd, err := requestFromSnapCmd(st, r); err != nil {
		return InternalError("cannot check request source")
	} else  if !fromSnapCmd {
		return Forbidden("only snap command can record notices")
	}
```

Reply:
updated